packages/truffle-core/lib/test.js

```diff
@@ -2,10 +2,11 @@ var Mocha = require("mocha");
 var chai = require("chai");
 var path = require("path");
 var Web3 = require("web3");
+var LegacyWeb3 = require("legacy-web3");
```

Comment:
Ah this `legacy-web3` again. Now that this effort is looking more like an actual, maintainable, possibly long-term solution, I think stuff like this is worth avoiding. We shouldn't have to mirror the package just cause we want to avoid a name collision. Why not instead leverage Node itself and group legacy stuff into separate packages that define specific version dependencies?